src/weather_gen.cpp

```diff
@@ -173,57 +174,57 @@ weather_type weather_generator::get_weather_conditions( const tripoint &location
 {
     w_point w( get_weather( location, t, seed ) );
     weather_type wt = get_weather_conditions( w );
-    // Make sure we don't say it's sunny at night! =P
-    if( wt == WEATHER_SUNNY && is_night( t ) ) {
-        return WEATHER_CLEAR;
-    }
     return wt;
 }
 
 weather_type weather_generator::get_weather_conditions( const w_point &w ) const
 {
-    weather_type r( WEATHER_CLEAR );
-    if( w.pressure > 1020 && w.humidity < 70 ) {
-        r = WEATHER_SUNNY;
-    }
-    if( w.pressure < 1010 && w.humidity > 40 ) {
-        r = WEATHER_CLOUDY;
-    }
-    if( r == WEATHER_CLOUDY && ( w.humidity > 96 || w.pressure < 1003 ) ) {
-        r = WEATHER_LIGHT_DRIZZLE;
-    }
-    if( r >= WEATHER_LIGHT_DRIZZLE && ( w.humidity > 97 || w.pressure < 1000 ) ) {
-        r = WEATHER_DRIZZLE;
-    }
-    if( r >= WEATHER_CLOUDY && ( w.humidity > 98 || w.pressure < 993 ) ) {
-        r = WEATHER_RAINY;
-    }
-    if( r == WEATHER_RAINY && w.pressure < 996 ) {
-        r = WEATHER_THUNDER;
-    }
-    if( r == WEATHER_THUNDER && w.pressure < 990 ) {
-        r = WEATHER_LIGHTNING;
-    }
+    weather_type current_conditions( WEATHER_DEFAULT );
+    for( int weather_index = WEATHER_DEFAULT; weather_index < weather::get_weather_count();
+         weather_index++ ) {
+
+        weather_requirements requires = weather::requirements( weather_index );
+        bool test_pressure =
+            requires.pressure_max > w.pressure &&
+            requires.pressure_min < w.pressure;
+        bool test_humidity =
+            requires.humidity_max > w.humidity &&
+            requires.humidity_min < w.humidity;
+        if( ( requires.humidity_and_pressure && !( test_pressure && test_humidity ) ) ||
+            ( !requires.humidity_and_pressure && !( test_pressure || test_humidity ) ) ) {
+            continue;
+        }
+        bool test_temperature =
+            requires.temperature_max > w.temperature &&
+            requires.temperature_min < w.temperature;
+        bool test_windspeed =
+            requires.windpower_max > w.windpower &&
+            requires.windpower_min < w.windpower;
+        bool test_acidic = !requires.acidic || w.acidic;
+        if( !( test_temperature && test_windspeed && test_acidic ) ) {
+            continue;
+        }
 
-    if( w.temperature <= 32 ) {
-        if( r == WEATHER_DRIZZLE ) {
-            r = WEATHER_FLURRIES;
-        } else if( r > WEATHER_DRIZZLE ) {
-            if( r >= WEATHER_THUNDER && w.windpower > 15 ) {
-                r = WEATHER_SNOWSTORM;
-            } else {
-                r = WEATHER_SNOW;
+        bool test_required_weathers = requires.required_weathers.empty();
+        if( !test_required_weathers ) {
+            std::string current_weather = weather::name( current_conditions );
```

Comment:
`weather::name` returns a translated name. Comparing it with the untranslated strings in `required_weathers` wont always work. I suggest you define and use ids for the weather types.